app/container.php

```diff
@@ -66,26 +62,52 @@ function (ConsoleCommandEvent $event) use ($userInteraction) {
 $container['process.silent_runner'] = function () {
     return new SilentProcessRunner();
 };
+
 $container['compose.executable_finder'] = function () {
     return new ComposeExecutableFinder();
 };
-$container['installer.docker'] = function ($c) {
-    return new DockerInstaller(
-        new \SRIO\ChainOfResponsibility\ChainBuilder([
-            new Homebrew($c['console.user_interaction'], $c['process.interactive_runner']),
-            new BrewCask($c['console.user_interaction'], $c['process.interactive_runner']),
-            new PhpSsh($c['console.user_interaction'], $c['process.interactive_runner']),
-            new Dinghy(new Boot2DockerCli($c['process.interactive_runner']), $c['cli.dinghy'], $c['console.user_interaction'], $c['process.interactive_runner']),
-            new DockerRouting($c['cli.dinghy'], $c['console.user_interaction'], $c['process.interactive_runner']),
-            new DnsDock(new SshClient(new Session(
+
+$container['installer.task_providers'] = function ($c) {
```

Comment:
Actually, I don't think that all tasks need to be instantiated each time... That way, we are creating all tasks instances for all operating system, it shouldn't be the case.

I'm not against additional specific containers per platform for the installation tasks or even operating system resolving directly in that closure.